scipy/special/tests/test_cdflib.py

```diff
@@ -461,40 +460,6 @@ def test_chndtrix_gh2158():
                82.35640899964173, 84.45263768373256]
     assert_allclose(res, res_exp)
 
-@pytest.mark.xfail_on_32bit("32bit fails due to algorithm threshold")
```

Comment:
Why are we removing these test cases? We're they failing because the old cdflib reference is inaccurate for some of them? The reference values could be replaced with `mpmath` then. This seems like a good opportunity to remove the `xfail_on_32bit` and show that they are still passing.

Reply:
Yes, the old reference values were very inaccurate for some cases. Allright, I will integrate them in the new tests against the mpmath reference. I might make a subjective selection though if that's okay with you.

Reply:
For high noncentrality values of 38 such as in the test originally, the mpmath reference computes CDF values of the order of `e-310`. There also boost fails to get close or returns nan, so I dropped these for noncentrality of 15 instead.

Reply:
Cases like this where the reference output is subnormal are why it makes sense to use an `atol` in tests. In cases like this, it's fine to return zero and signal underflow through `sf_error`. Is Boost returning `nan` where the previous method used to return 0? I'd count that as a bug.

Reply:
Fully agree that the `nan` feels like a bug.

Reply:
@dschmitz89, do you plan to follow up on these `nan`s?

Reply:
@steppi : I do , am just short on time at the moment. Not sure I can make it till the 1.15 release as I am also going on holiday at the end of the month. I want to spend my scipy activities until then mostly on review. If you can/want to take over the last todos here, that would be much appreciated as I think this PR is almost ready to merge.

Reply:
@steppi : I just had another look at the test cases for `nc=38`. Even for small values within the normal floating point range, we get high discrepancies.

For `df, nc, t = 0.98, 38, 1.5` the mpmath reference computes `1.546906350750292e-295` but boost results in `2.5919953604832226e-97`. Would you prefer to get the test to pass with something like `atol=1e-90`? I would simply ignore these edge cases for now to be honest. In the parameter ranges that matter most we get much better results with boost.

Reply:
It looks like you set `mp.dps` too low to evaluate the sum. I set `mp.dps=500` and got the result `2.591995360483094e-97` from your reference implementation. Much closer to Boost. If you're evaluating a sum with `mpmath`, you may need to set `dps` pretty high.

Reply:
Ouch, thanks for spotting this. With `mp.dps=500`, the reference calculation did not finish within 10 minutes on my laptop though. It has performance issues at the moment but this is crazy. How long does it take on your machine? Might try out my Mac instead.

Reply:
With 400 significant digits I also get `2.591995360483094e-97` within 12 minutes of computation time.